pkg/scheduler/api/device_info.go

```diff
@@ -55,15 +55,27 @@ func (g *GPUDevice) getUsedGPUMemory() uint {
 
 // GetGPUResourceOfPod returns the GPU resource required by the pod.
 func GetGPUResourceOfPod(pod *v1.Pod) uint {
+	var initMem uint
+	for _, container := range pod.Spec.InitContainers {
+		res := getGPUResourceOfContainer(container)
+		if initMem < res {
+			initMem = res
+		}
+	}
+
 	var mem uint
 	for _, container := range pod.Spec.Containers {
-		mem += getGPUResourceOfContainer(&container)
+		mem += getGPUResourceOfContainer(container)
 	}
-	return mem
+
+	if mem > initMem {
+		return mem
+	}
+	return initMem
 }
 
 // getGPUResourceOfPod returns the GPU resource required by the container.
-func getGPUResourceOfContainer(container *v1.Container) uint {
+func getGPUResourceOfContainer(container v1.Container) uint {
```

Comment:
Suggest to keep the object reference as the input parameter for we only need the `spec.resource` part. That will be more efficient.

Reply:
Make sense, currently we pass a value here.